buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -59,8 +68,10 @@ class VagrantTestPlugin implements Plugin<Project> {
     @Override
     void apply(Project project) {
 
+        collectAvailableBoxes(project)
```

Comment:
Since this is static maybe you should do it in a static block instead of on plugin application.

Reply:
Sure that makes sense. Actually general question about that (static block vs static method). I've always erred on the side of not doing stuff in static blocks when it could conceivably blow up (not sure if that applies here) because it seems more intuitive for it to blow up when it's used (in this case when the plugin is applied) than when the class is loaded. Does that make sense or is that just a weird preconception I have

Reply:
It makes some sense, yeah. static blocks are executed when the class is loaded but something like this is executed on every plugin apply. Minimally, the stack trace of the second one is going to be more useful than the first. You could certainly do the static stuff on *first* plugin apply, but that feels a little more complicated than it is worth in this scenario.

Reply:
That's a good point. I'll put it in a static block since it's applicable to all applications of the plugin, in contrast to some of the other configuration here which uses properties of the project

Reply:
This method is now non-static because of some changes I made to avoid separate plugin instances stomping on each others' configuration